integration/gateway/gateway_suite_test.go

```diff
@@ -0,0 +1,169 @@
+}, func() {
+	buildServer.Shutdown()
+})
+
+func StartPort() int {
+	return integration.GatewayBasePort.StartPortForNode()
+}
+
+func grpcClient(tlsRootCertFile string) *comm.GRPCClient {
+	caPEM, res := ioutil.ReadFile(tlsRootCertFile)
+	Expect(res).To(BeNil())
+
+	gClient, err := comm.NewGRPCClient(comm.ClientConfig{
+		Timeout: 3 * time.Second,
+		KaOpts:  comm.DefaultKeepaliveOptions,
+		SecOpts: comm.SecureOptions{
+			UseTLS:        true,
+			ServerRootCAs: [][]byte{caPEM},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(gClient).NotTo(BeNil())
+
+	return gClient
+}
+
+func GatewayClient(address, tlsRootCertFile string) gateway.GatewayClient {
+	grpcClient := grpcClient(tlsRootCertFile)
+	conn, err := grpcClient.NewConnection(address)
+	Expect(err).NotTo(HaveOccurred())
+	gatewayClient := gateway.NewGatewayClient(conn)
+
+	return gatewayClient
+}
+
+func NewProposedTransaction(signingIdentity msp.SigningIdentity, channelName, chaincodeName, transactionName string, args ...[]byte) *gateway.ProposedTransaction {
+	proposal, txnID := newProposalProto(signingIdentity, channelName, chaincodeName, transactionName, args...)
+	signedProposal, err := protoutil.GetSignedProposal(proposal, signingIdentity)
+	Expect(err).NotTo(HaveOccurred())
+
+	txn := &gateway.ProposedTransaction{
+		Proposal:  signedProposal,
+		TxId:      txnID,
+		ChannelId: channelName,
+	}
+
+	return txn
+}
+
+func newProposalProto(signingIdentity msp.SigningIdentity, channelName, chaincodeName, transactionName string, args ...[]byte) (*peer.Proposal, string) {
+	creator, err := signingIdentity.Serialize()
+	Expect(err).NotTo(HaveOccurred())
+
+	invocationSpec := &peer.ChaincodeInvocationSpec{
+		ChaincodeSpec: &peer.ChaincodeSpec{
+			Type:        peer.ChaincodeSpec_NODE,
+			ChaincodeId: &peer.ChaincodeID{Name: chaincodeName},
+			Input:       &peer.ChaincodeInput{Args: chaincodeArgs(transactionName, args...)},
+		},
+	}
+
+	result, transactionID, err := protoutil.CreateChaincodeProposal(
+		common.HeaderType_ENDORSER_TRANSACTION,
+		channelName,
+		invocationSpec,
+		creator,
+	)
+	Expect(err).NotTo(HaveOccurred())
+
+	return result, transactionID
+}
+
+func chaincodeArgs(transactionName string, args ...[]byte) [][]byte {
+	result := make([][]byte, len(args)+1)
+
+	result[0] = []byte(transactionName)
+	copy(result[1:], args)
+
+	return result
+}
+
+// TODO not the only suite doing this - move this to nwo?
+func GetSigningIdentity(network *nwo.Network, org, user, mspID, mspType string) msp.SigningIdentity {
+	peerForOrg := network.Peer(org, "peer0")
+	mspConfigPath := network.PeerUserMSPDir(peerForOrg, user)
+	mspInstance := loadLocalMSPAt(mspConfigPath, mspID, mspType)
+
+	signingIdentity, err := mspInstance.GetDefaultSigningIdentity()
+	Expect(err).NotTo(HaveOccurred())
+	return signingIdentity
+}
+
+// TODO not the only suite doing this - move this to nwo?
+func loadLocalMSPAt(dir, id, mspType string) msp.MSP {
+	Expect(mspType).To(Equal("bccsp"))
```

Comment:
No, instead I'd ask why we're using an MSP and BCCSP in these tests. I expect there are some good use cases but, for this one, the cert is the serialized identity and the key is the signer.

Put differently, we need this:
```
// Signer is the interface needed to sign a transaction
type Signer interface {
	Sign(msg []byte) ([]byte, error)
	Serialize() ([]byte, error)
}
```
and this is building
```

// Identity interface defining operations associated to a "certificate".
// That is, the public part of the identity could be thought to be a certificate,
// and offers solely signature verification capabilities. This is to be used
// at the peer side when verifying certificates that transactions are signed
// with, and verifying signatures that correspond to these certificates.///
type Identity interface {

	// ExpiresAt returns the time at which the Identity expires.
	// If the returned time is the zero value, it implies
	// the Identity does not expire, or that its expiration
	// time is unknown
	ExpiresAt() time.Time

	// GetIdentifier returns the identifier of that identity
	GetIdentifier() *IdentityIdentifier

	// GetMSPIdentifier returns the MSP Id for this instance
	GetMSPIdentifier() string

	// Validate uses the rules that govern this identity to validate it.
	// E.g., if it is a fabric TCert implemented as identity, validate
	// will check the TCert signature against the assumed root certificate
	// authority.
	Validate() error

	// GetOrganizationalUnits returns zero or more organization units or
	// divisions this identity is related to as long as this is public
	// information. Certain MSP implementations may use attributes
	// that are publicly associated to this identity, or the identifier of
	// the root certificate authority that has provided signatures on this
	// certificate.
	// Examples:
	//  - if the identity is an x.509 certificate, this function returns one
	//    or more string which is encoded in the Subject's Distinguished Name
	//    of the type OU
	// TODO: For X.509 based identities, check if we need a dedicated type
	//       for OU where the Certificate OU is properly namespaced by the
	//       signer's identity
	GetOrganizationalUnits() []*OUIdentifier

	// Anonymous returns true if this is an anonymous identity, false otherwise
	Anonymous() bool

	// Verify a signature over some message using this identity as reference
	Verify(msg []byte, sig []byte) error

	// Serialize converts an identity to bytes
	Serialize() ([]byte, error)

	// SatisfiesPrincipal checks whether this instance matches
	// the description supplied in MSPPrincipal. The check may
	// involve a byte-by-byte comparison (if the principal is
	// a serialized identity) or may require MSP validation
	SatisfiesPrincipal(principal *msp.MSPPrincipal) error
}

// SigningIdentity is an extension of Identity to cover signing capabilities.
// E.g., signing identity should be requested in the case of a client who wishes
// to sign transactions, or fabric endorser who wishes to sign proposal
// processing outcomes.
type SigningIdentity interface {

	// Extends Identity
	Identity

	// Sign the message
	Sign(msg []byte) ([]byte, error)

	// GetPublicVersion returns the public parts of this identity
	GetPublicVersion() Identity
}
```

Reply:
@jt-nti PR https://github.com/hyperledger/fabric/pull/2408 will help address this, I think

Reply:
Aha, #2408 seems like what I was looking for, I'll give it a go. Thanks @sykesm @andrew-coleman